tools/lsp/preview/properties.rs

```diff
@@ -92,11 +97,11 @@ fn add_element_properties(
             return None;
         }
 
-        let declared_at = value
-            .type_node()
-            .as_ref()
-            .and_then(util::map_node_and_url)
-            .map(|(uri, range)| DeclarationInformation { uri, start_position: range.start });
+        let declared_at = value.type_node().as_ref().map(|n| DeclarationInformation {
+            uri: Url::from_file_path(n.source_file.path())
+                .unwrap_or_else(|_| Url::parse("file:///unnamed").unwrap()),
```

Comment:
I don't like this. This creates a file name that is not existing. 
It might be better if the `DeclarationInformation::uri` was a either `Option<Url>`, or a PathBuf (mapping to the type used in the compiler).

Reply:
In practice I do not think the conversion can fail: We convert Url to PathBuf right when getting input from the LSP and basically reject anything that does not convert in that place. In general we map from PathBuf to URL and back way too often, and I am sure we just unwrap() in some places.

IMHO we should just use URLs in the compiler as well: We do have paths like "builtin://foo", so we apparently want URLs there as well :-) Then there are resource URLs in the mix as well... which I have found in PathBufs before.

Reply:
Sure, but this "unnamed" in the code now looks terrible.
I think in this case since this is not meant to be sent through the LSP, PathBuf is better until we change the compiler to use another type for its filenames.

Reply:
Added a new patch changing this member from Url to PathBuf.